lib/sequent/core/event_store.rb

```diff
@@ -250,8 +253,12 @@ def store_events(command, streams_with_events = [])
             Sequent::Core::Oj.dump(events),
           ],
         )
-      rescue ActiveRecord::RecordNotUnique
-        raise OptimisticLockingError
+      rescue ActiveRecord::RecordNotUnique => e
+        if e.message =~ /duplicate aggregate key value/
+          raise AggregateKeyNotUniqueError
```

Comment:
Can we add here what value is not unique?

Reply:
Not easily. We'd have to insert the unique keys one by one to and then somehow include it in the SQL exception message and parse it in the Ruby code. I think for 99% of the cases it will just be an aggregate with a single constraint (e.g. user with an email login) so the unique violation is mostly clear.

Of course, in more complex cases it might be useful for debugging. We can at least include the SQL error message in the exception message I guess, or a caused by?

Reply:
Ruby will automatically add the causing exception as `cause` of `AggregateKeyNotUniqueError` in this scenario.

Reply:
Added at least the aggregate type and id to the error message.